### PR TITLE
DIADO-2683 CIS Hardening of Amazon Linux 2 based on Benchmark v1.0.0

### PR DESCRIPTION
**Pull request by 60k41p**
Sections 4, 5, 6
Documentation updates

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -6,38 +6,57 @@
...
 # If false, a debug message will be generated instead, with the preface *** ACTION REQUIRED ***..
 fail_on_manual_remediation_actions: false  # True or false.
 
+cis_target_os_distribution: "Amazon"
+cis_target_os_versions:
+  - "2016.03"
+  - "2016.09"
+  - "2017.03"
+  - "2017.09"
+  - "2017.12"
```
Comment (stephenmuss) [inline defaults/main.yml:29]: Should `2018.03` be here?

File: defaults/main.yml [hunk 2]
```diff
@@ -46,97 +65,89 @@ cis_aide_cron_dow: "*" # Day of week
...
 cis_sshd_max_auth_tries: 4
-
-# 5.2.11
 cis_sshd_ciphers: "aes256-ctr,aes192-ctr,aes128-ctr"
-
-# 5.2.12
 cis_sshd_macs: "hmac-sha2-512-etm@openssh.com,hmac-sha2-256-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-512,hmac-sha2-256,umac-128@openssh.com"
-
-# 5.2.13
+cis_sshd_kex_algos: "curve25519-sha256,curve25519-sha256@libssh.org,diffie-hellman-group14-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,ecdh-sha2-nistp521,ecdh-sha2-nistp384,ecdh-sha2-nistp256,diffie-hellman-group-exchange-sha256"
```
Comment (stephenmuss) [inline defaults/main.yml:108]: Can we make this `cis_sshd_kex_algorithms`?

File: meta/main.yml [hunk 3]
```diff
@@ -15,5 +15,6 @@ galaxy_info:
     - "2017.09"
     - "2017.12"
```
Comment (stephenmuss) [inline meta/main.yml:16]: 2018.03 here too?

File: tasks/level-1/5.3.1.yml [hunk 4]
```diff
@@ -3,6 +3,24 @@
...
 # 5.3.1 Ensure password creation requirements are configured
 
+- name: 5.3.1 - Ensure password creation requirements are honoured by PAM
+  pamd:
+    name: "{{ item }}"
+    type: password
+    control: requisite
+    module_path: pam_pwquality.so
+    module_arguments: 'try_first_pass
+        retry=3'
```
Comment (stephenmuss) [inline tasks/level-1/5.3.1.yml:13]: Are these two lines correct? They look funky.

File: tasks/level-1/5.5.yml [hunk 5]
```diff
@@ -1,26 +1,6 @@
...
-    line: "wheel:x:10:{{ cis_wheel_group_members }}"
-    dest: "/etc/group"
-  tags:
-    - level-1
-    - section-5
-    - "5.5"
-    - scored
+# 5.5 Ensure root login is restricted to system console
+# Not implemented, because of potential impact to su -, and we do not know which consoles are in physically secure locations
+# Please tune /etc/secretty manually per site circumstances
```
Comment (stephenmuss) [inline tasks/level-1/5.5.yml:6]: Is it worth leaving the contents of this file here but commented out?

Perhaps you could use `debug` to output that comment you have written too.

File: defaults/main.yml (hunk 1 above)
Comment (60k41p) [reply defaults/main.yml:29]: If anything we should remove all entries but Karoo - we no longer target Amazon Linux 1.
Karoo would cover all point releases of Amazon Linux 2.

File: defaults/main.yml (hunk 2 above)
Comment (60k41p) [reply defaults/main.yml:108]: noted. Will do after the merge.

File: meta/main.yml (hunk 3 above)
Comment (60k41p) [reply meta/main.yml:16]: Same as above, I think we should only leave Karoo there. Thanks for bringing this up.

File: tasks/level-1/5.3.1.yml (hunk 4 above)
Comment (60k41p) [reply tasks/level-1/5.3.1.yml:13]: they do look funky. They work as intended - I have verified by looking at the resulting config files. Similar format in 5.3.2.

Ansible docs say that parameter is a list. I will convert these lines to list format after the merge.

File: tasks/level-1/5.5.yml (hunk 5 above)
Comment (60k41p) [reply tasks/level-1/5.5.yml:6]: yes, will do.

File: defaults/main.yml (hunk 1 above)
Comment (stephenmuss) [reply defaults/main.yml:29]: I agree that AL1 releases should go. This is specifically targeting AL2.

**Review by stephenmuss — APPROVED** [review]

**Pull request merged**